test/unittest/conftest.py

```diff
@@ -117,247 +56,116 @@ def an_insert(float_tensors_32):
-    float_tensors_32.add(
-        TorchModel(
-            object=torch.nn.Linear(32, 16),
-            identifier='linear_a_base',
-            encoder='torch.float32[16]',
-            preprocess=lambda r: r['x'],
-        ),
-    )
-    yield float_tensors_32
-    try:
-        float_tensors_32.remove('model', 'linear_a_base', force=True)
-    except TypeError as e:
-        if "'NoneType' object is not subscriptable" in str(e):
-            return
-        raise e
-
-
-@pytest.fixture()
-def a_listener(a_model):
-    a_model.distributed = False
-    a_model.add(
-        Listener(
-            model='linear_a',
-            select=Collection('documents').find(),
-            key='x',
-        )
-    )
-    yield a_model
-    a_model.remove('listener', 'linear_a/x', force=True)
+@pytest.fixture
+def local_empty_data_layer(request) -> Datalayer:
+    _data_layer = build_datalayer(CFG, data_backend=MONGOMOCK_URI)
+    return _data_layer
 
 
-@pytest.fixture()
-def a_listener_base(a_model_base):
-    a_model_base.add(
-        Listener(
-            model='linear_a_base',
-            select=Collection('documents').find({}, {'_id': 0}),
-            key='_base',
-        )
-    )
-    yield a_model_base
-    a_model_base.remove('listener', 'linear_a_base/_base', force=True)
+@pytest.fixture
+def empty_collection() -> Collection:
+    return Collection(str(uuid.uuid4()))
 
 
-@pytest.fixture()
-def a_classifier(float_tensors_32):
-    float_tensors_32.add(
-        TorchModel(
-            object=BinaryClassifier(32),
-            identifier='classifier',
-        ),
-    )
-    yield float_tensors_32
-    try:
-        float_tensors_32.remove('model', 'classifier', force=True)
-    except TypeError as e:
-        if "'NoneType' object is not subscriptable" in str(e):
-            return
-        raise e
-
-
-@pytest.fixture()
-def b_model(float_tensors_32, float_tensors_16, float_tensors_8):
-    float_tensors_32.add(
-        TorchModel(
-            object=torch.nn.Linear(16, 8),
-            identifier='linear_b',
-            encoder='torch.float32[8]',
-        ),
-    )
-    yield float_tensors_32
-    try:
-        float_tensors_32.remove('model', 'linear_b', force=True)
-    except TypeError as e:
-        if "'NoneType' object is not subscriptable" in str(e):
-            return
-        raise e
-
-
-@pytest.fixture()
-def c_model(float_tensors_32, float_tensors_16):
-    float_tensors_32.add(
-        TorchModel(
-            object=torch.nn.Linear(32, 16),
-            identifier='linear_c',
-            encoder='torch.float32[16]',
-        ),
-    )
-    yield float_tensors_32
-    try:
-        float_tensors_32.remove('model', 'linear_c', force=True)
-    except TypeError as e:
-        if "'NoneType' object is not subscriptable" in str(e):
-            return
-        raise e
+@pytest.fixture
+def local_collection_with_random_data(data_layer: Datalayer, request) -> Collection:
```

Comment:
I think we should have only 
local and global data_layer fixture.

Like, local and global data_layer both already have a random data added collection in them.

But I see this creates a new collection upon each request and fills it with random data.

can we remove this fixture, if not im fine either :)

Reply:
I had deleted it